### ISSUE TITLE
[a11y] Completed state badges not conveyed in High Contrast

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#1749]
## Issue summary

State badge icons for “complete” tasks are not conveyed correctly in Windows High Contrast since the filled state relies on a background color alone, which is stripped out and replaced by user settings in High Contrast (The foreground color is ignored, so the background color that relies on it is as well). ([1.4.1 Use of Color](https://www.w3.org/TR/WCAG21/#use-of-color), Level A).

## Expected behavior

The unfilled, partially filled, filled state will be visible in High Contrast.

## Actual behavior

Filled states look the same as unfilled states.

<img width="900" alt="Default filled state for a gray badge" src="https://user-images.githubusercontent.com/1462085/60216397-c7cc3300-981e-11e9-8985-88ea196fb95a.png">

<img width="603" alt="Default filled state for a gray badge in High Contrast" src="https://user-images.githubusercontent.com/1462085/60216799-db2bce00-981f-11e9-9f36-260ee45c7fab.png">

```CSS
.Polaris-Badge--progressComplete .Polaris-Badge__Pip {
    background: currentColor;
}

.Polaris-Badge .Polaris-Badge__Pip {
    color: #919eab;
}
```

## Steps to reproduce the problem

1. Navigate to the badge examples in any supported browser: https://polaris.shopify.com/components/images-and-icons/badge
1. View the "complete" badge example
1. Open the badge examples page in Edge on Windows 10 using any [Windows High Contrast theme](https://support.microsoft.com/en-us/help/4026951/windows-10-turn-high-contrast-mode-on-or-off)

## Recommendation

Changing the pip to use a linear background gradient (like the partial style) instead of a solid background color. 🤷‍♀This tricks Windows into treating the background style like a background _image_, which is ignores.

Partial style:

```CSS
.Polaris-Badge--progressPartiallyComplete .Polaris-Badge__Pip {
    background: linear-gradient(0deg,currentColor,currentColor 50%,transparent 0,transparent);
}

.Polaris-Badge .Polaris-Badge__Pip {
    color: #919eab;
}
```

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.17.0+
- Browser: Edge
- Device: Windows VM
- Operating System: Windows 10

**Comment by dpersing** [comment]
Hi @chloerice! Any updates on this issue?

**Comment by chloerice** [comment]
Hey @dpersing, pairing on this would be really helpful! I spent last Wednesday trying to get VirtualBox set up on my machine, but it made my disk full like Parallels did so I'm unable to use Windows still 😞
